### PR TITLE
Bluetooth: Mesh: Fixes wrong prov process

### PR DESCRIPTION
**Pull request by LingaoM**
When link close, Wrong clear pointer, resulting
in illegal pointer access.

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

### CODE REVIEW

**Review by trond-snekvik — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Comment by trond-snekvik** [comment]
Sorry, github was only showing me the strdup change, I need to take a second look

File: subsys/bluetooth/mesh/pb_adv.c [hunk 1]
```diff
@@ -460,7 +460,7 @@ static void gen_prov_start(struct prov_rx *rx, struct net_buf_simple *buf)
 	link.rx.last_seg = START_LAST_SEG(rx->gpc);
 
 	if ((link.rx.seg & BIT(0)) &&
-	    (find_msb_set(~link.rx.seg) >= link.rx.last_seg)) {
```
Comment (LingaoM) [inline subsys/bluetooth/mesh/pb_adv.c:463]: `find_msb_set(~link.rx.seg) `this always return **32** here.

File: subsys/bluetooth/mesh/prov.h [hunk 2]
```diff
@@ -106,11 +111,6 @@ struct bt_mesh_prov_link {
 	uint8_t conf_key[16];           /* ConfirmationKey */
 	uint8_t conf_inputs[145];       /* ConfirmationInputs */
 	uint8_t prov_salt[16];          /* Provisioning Salt */
-
-	const struct prov_bearer *bearer;
-	const struct bt_mesh_prov_role *role;
```
Comment (LingaoM) [inline subsys/bluetooth/mesh/prov.h:111]: Wrong clear `bearer `and `role` pointer.

File: subsys/bluetooth/mesh/pb_adv.c (hunk 1 above)
Comment (trond-snekvik) [reply subsys/bluetooth/mesh/pb_adv.c:463]: Replace `& 0xff` with `& SEG_NVAL`? It's not clear from this line that this is the reason we need 0xff.

**Review by trond-snekvik — COMMENTED** [review]
The changes all look correct to me, but the first two commits could use some better commit messages. It's not clear from the commit messages what the actual problems are, and it shouldn't be necessary to go the the PR in github to see the explanation in the comments

**Review by jhedberg — APPROVED** [review]

**Review by trond-snekvik — APPROVED** [review]

**Pull request merged**